### PR TITLE
Add required config changes

### PR DESCRIPTION
**Pull request by kaituo**
Note: since there are a lot of dependencies, I only list the main class and test code to save reviewers' time. The build will fail due to missing dependencies. I will use that PR just for review. will not merge it. Will have a big one in the end and merge once after all review PRs get approved.

*Issue #, if available:*

*Description of changes:*

This PR adds config changes required for multi-entity detectors.

Testing done:
1. Manually tested nothing breaks with these changes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorPlugin.java [hunk 1]
```diff
@@ -343,6 +360,86 @@ private static Void initGson() {
...
+            AnomalyDetectorSettings.MIN_PREVIEW_SIZE,
+            AnomalyDetectorSettings.HOURLY_MAINTENANCE,
+            AnomalyDetectorSettings.HOURLY_MAINTENANCE,
+            entityColdStarter,
+            modelPartitioner,
+            featureManager,
+            memoryTracker
+        );
+
+        // EntityCache cache = new LRUCache(
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorPlugin.java:415]: Remove these commented code ?

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java [hunk 2]
```diff
@@ -223,13 +225,39 @@ private AnomalyDetectorSettings() {}
...
 
     // how many categorical fields we support
     public static final int CATEGORY_FIELD_LIMIT = 1;
 
+    public static final int MULTI_ENTITY_NUM_TREES = 10;
+
+    // cache related
+    public static final int DEDICATED_CACHE_SIZE = 10;
+
+    // We only keep priority (4 bytes float) in inactive cache. 1 million priorities
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java:248]: So we store active entity's priority in inactive entity's cache ? If that's the case, why not name it as MAX_ENTITIES ?

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorPlugin.java (hunk 1 above)
Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/AnomalyDetectorPlugin.java:415]: yeah, removed

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java [hunk 3]
```diff
@@ -239,4 +267,17 @@ private AnomalyDetectorSettings() {}
...
+        .intSetting("opendistro.anomaly_detection.max_primary_shards", 10, 0, Setting.Property.NodeScope, Setting.Property.Dynamic);
+
+    // max entity value's length
+    public static int MAX_ENTITY_LENGTH = 256;
+
+    // max number of index checkpoint requests in one bulk
+    public static int MAX_BULK_CHECKPOINT_SIZE = 1000;
+
+    // number of bulk checkpoints per second
+    public static double CHECKPOINT_BULK_PER_SECOND = 0.02;
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java:282]: Can execute 1 bulk index checkpoint every 50 seconds?

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java (hunk 2 above)
Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java:248]: we don't do that.  Active entities' states has both models and priority, while inactive entities' states has only priority.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java (hunk 3 above)
Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java:282]: yeah,  1/60 is not a finite decimal representation.  It is fine.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java (hunk 2 above)
Comment (ylwu-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java:248]: Got it, misunderstanding, thought all priority of active&inactive entity will be stored in inactive cache.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java [hunk 4]
```diff
@@ -223,13 +225,39 @@ private AnomalyDetectorSettings() {}
...
+    public static final int MULTI_ENTITY_NUM_TREES = 10;
+
+    // cache related
+    public static final int DEDICATED_CACHE_SIZE = 10;
+
+    // We only keep priority (4 bytes float) in inactive cache. 1 million priorities
+    // take up 4 MB.
+    public static final int MAX_INACTIVE_ENTITIES = 1_000_000;
+
+    // TODO: check how much does 1 million insertion costs in memory
```
Comment (saratvemulapalli) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java:252]: Do we still have to do a performance test get the accurate number?

**Review by saratvemulapalli — APPROVED** [review]
The changes look good to me.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java (hunk 4 above)
Comment (kaituo) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/settings/AnomalyDetectorSettings.java:252]: seems roughly 1 MB per 1 million from Weicong's jmap data.

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change

**Pull request closed without merging**